### PR TITLE
Fix CI, set new minimal rust version

### PR DESCRIPTION
**Pull request by luke-biel**
Can you take a look if it's all ok with you @frondeus ?

### CODE REVIEW

File: acceptance_tests/basic/src/lib.rs [hunk 1]
```diff
@@ -147,17 +147,17 @@ mod test_cases {
     #[derive(Debug)]
     #[allow(dead_code)]
     enum SimpleEnum {
-        Var1(i32),
```
Comment (frondeus) [inline acceptance_tests/basic/src/lib.rs:150]: Why did you touch this enum?

File: acceptance_tests/hamcrest_assertions/src/lib.rs [hunk 2]
```diff
@@ -17,6 +17,6 @@ mod test_cases {
     #[test_case(&[2, 4] => it contains(vec!(2, 4)))]
     #[test_case(&[2, 3] => is len(1))]
     fn removes_odd_numbers(collection: &[u8]) -> &Vec<u8> {
-        Box::leak(Box::new(collection.into_iter().filter(|x| *x % 2 == 0).copied().collect()))
+        Box::leak(Box::new(collection.into_iter().filter(|x| *x % 2 == 0).map(|v| *v).collect()))
```
Comment (frondeus) [inline acceptance_tests/hamcrest_assertions/src/lib.rs:20]: I think `copied` was ok, why did you change it?

Comment (luke-biel) [reply acceptance_tests/hamcrest_assertions/src/lib.rs:20]: It was stabilised in `1.36` - if we wish to support versions below that, we have to work around it.

File: acceptance_tests/basic/src/lib.rs (hunk 1 above)
Comment (luke-biel) [reply acceptance_tests/basic/src/lib.rs:150]: I had to due to `quote!` and `Debug` combination producing different outputs on different rust versions for ADTs.
Right now we have that space between `Var` and `(_, 5)` that we test for but it is not there for `1.41` and above.

Comment (luke-biel) [reply acceptance_tests/basic/src/lib.rs:150]: I can't think of a way to test for ADTs with `panic` keyword sadly, until I can guarantee that the message produced will have consistent text across versions.
